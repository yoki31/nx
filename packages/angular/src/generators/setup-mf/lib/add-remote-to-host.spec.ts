import { tsquery } from '@phenomnomnominal/tsquery';
import type { ArrayLiteralExpression } from 'typescript';
import { checkIsCommaNeeded } from './add-remote-to-host';

describe('Add remote to host', () => {
  it('should add remote correctly even in multiline', () => {
    // ARRANGE
    const hostWebpackConfig = `const obj = {
            remotes: [
                'remote1',
                'remote2',
            ]
        }`;
    const webpackAst = tsquery.ast(hostWebpackConfig);
    const mfRemotesNode = tsquery(
      webpackAst,
      'Identifier[name=remotes] ~ ArrayLiteralExpression',
      { visitAllChildren: true }
    )[0] as ArrayLiteralExpression;

    const endOfPropertiesPos = mfRemotesNode.getEnd() - 1;

    // ACT
    const isCommaNeeded = checkIsCommaNeeded(mfRemotesNode.getText());

    const updatedConfig = `${hostWebpackConfig.slice(0, endOfPropertiesPos)}${
      isCommaNeeded ? ',' : ''
    }'remote3',${hostWebpackConfig.slice(endOfPropertiesPos)}`;

    // ASSERT
    expect(updatedConfig).toMatchSnapshot();
  });
});
