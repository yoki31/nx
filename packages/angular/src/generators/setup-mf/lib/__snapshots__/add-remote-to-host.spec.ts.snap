// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add remote to host should add remote correctly even in multiline 1`] = `
"const obj = {
            remotes: [
                'remote1',
                'remote2',
            'remote3',]
        }"
`;
